puffin/src/lib.rs

```diff
@@ -525,13 +630,34 @@ impl GlobalProfiler {
     pub fn add_sink(&mut self, sink: FrameSink) -> FrameSinkId {
         let id = self.next_sink_id;
         self.next_sink_id.0 += 1;
+
         self.sinks.insert(id, sink);
         id
     }
 
     pub fn remove_sink(&mut self, id: FrameSinkId) -> Option<FrameSink> {
         self.sinks.remove(&id)
     }
+
+    /// Insert custom scopes into puffin.
+    /// Scopes details should only be provided once for each scope and need be inserted before being reported to puffin.
+    /// This is only relevant when your not using puffin through the profiler macros.
+    pub fn insert_custom_scopes(scopes: &[ScopeDetails]) {
+        let mut lock = Self::lock();
+        let new_scopes = lock.scope_collection.register_custom_scopes(scopes);
+        lock.scope_delta.extend(&new_scopes);
+    }
+
+    /// Fetches and drains the delta of newly registered scopes if any.
+    /// Useful for knowing which scopes were registered since last time the function was called.
+    pub fn take_scope_delta(&mut self) -> Vec<ScopeId> {
+        std::mem::take(&mut self.scope_delta)
+    }
+
+    pub fn scope_collection() -> ScopeCollection {
+        let lock = Self::lock();
+        lock.scope_collection.clone()
+    }
```

Comment:
There are some issues with a sink-only approach, which I would prefer too. For example, in case of web-server. Webserver profiles, I connect with client, I'm adding a sync for this client on the server, I wont get any old deltas. The global profiler is the only place it seems I can objectively store what scope details we have and can provide them to any new sinks registered after the first scopes have been registered.